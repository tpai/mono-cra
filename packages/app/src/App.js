import React, { PureComponent } from 'react';
import * as UI from 'ui-base';

class App extends PureComponent {
  render() {
    return (
      <div>
        <UI.Button handleClick={() => alert('imported button clicked')}>
          Click Me in App
        </UI.Button>
      </div>
    );
  }
}

export default App;
