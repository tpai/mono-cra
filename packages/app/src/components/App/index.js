import React, { PureComponent } from 'react';
import cx from 'classnames';
import { Button } from 'ui-base';

import styles from './styles.module.scss';

class App extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      isPaused: false,
    };
  }
  handleToggleRotation = () => {
    this.setState(({ isPaused }) =>
      ({ isPaused: !isPaused })
    );
  }
  render() {
    const { isPaused } = this.state;
    return (
      <div className={styles.container}>
        <div
          className={cx(styles.logo, {
            [styles.running]: !isPaused,
            [styles.pause]: isPaused,
          })}
        />
        <Button
          className={cx(styles.button, {
            [styles.pause]: !isPaused,
          })}
          handleClick={this.handleToggleRotation}
        >
          <i className={cx('fas', isPaused ? 'fa-play' : 'fa-pause')}></i>
          {isPaused ? 'Start' : 'Stop'}
        </Button>
      </div>
    );
  }
}

export default App;
