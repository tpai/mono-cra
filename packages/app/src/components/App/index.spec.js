import React from 'react';
import App from 'components/App';

describe('components/App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should render initial component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correct component when rotation is paused', () => {
    wrapper.setState({ isPaused: true });
    expect(wrapper).toMatchSnapshot();
  });
});
