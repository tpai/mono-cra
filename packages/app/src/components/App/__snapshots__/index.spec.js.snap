// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/App should render correct component when rotation is paused 1`] = `
<div
  className="container"
>
  <div
    className="logo pause"
  />
  <Button
    className="button"
    handleClick={[Function]}
  >
    <i
      className="fas fa-play"
    />
    Start
  </Button>
</div>
`;

exports[`components/App should render initial component 1`] = `
<div
  className="container"
>
  <div
    className="logo running"
  />
  <Button
    className="button pause"
    handleClick={[Function]}
  >
    <i
      className="fas fa-pause"
    />
    Stop
  </Button>
</div>
`;
